src/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -32,7 +32,7 @@ internal override async Task<AuthenticationResult> ExecuteAsync(CancellationToke
                 var msalAccessTokenItem = await CacheManager.FindAccessTokenAsync().ConfigureAwait(false);
                 if (msalAccessTokenItem != null)
                 {
-                    return new AuthenticationResult(msalAccessTokenItem, null);
+                    return new AuthenticationResult(msalAccessTokenItem, null, AuthenticationRequestParameters.RequestContext.TelemetryCorrelationId);
```

Comment:
is this intentional?

Reply:
yes. the correlation ID needs to be made available on the AuthenticationResult so I pass it into the constructor

Reply:
as mentioned elsewhere... this is the telemetrycorrelationid and not the new id. Also why are you creating this as a new Guid?

---
In reply to: [297841976](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1235#discussion_r297841976) [](ancestors = 297841976)

Reply:
We have two names for the same value so this is a little confusing. the telemetrycorrelationid and the correlationid are the same. Also, I create a new guid because we are exposing the correlation id as a GUID on the AuthenticationResult but the correlation id in the requestContext is exposed as a string so i have to make the conversion.

---
In reply to: [298004949](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1235#discussion_r298004949) [](ancestors = 298004949,297841976)